numpy/core/tests/test_cpu_features.py

```diff
@@ -48,11 +57,72 @@ def assert_features_equal(actual, desired, fname):
         "%s"
     ) % (fname, actual, desired, error_report))
 
+def _text_to_list(txt):
+    out = txt.strip("][\n").replace("'", "").split(', ')
+    return None if out[0] == "" else out
+
 class AbstractTest:
     features = []
     features_groups = {}
     features_map = {}
     features_flags = set()
+    cwd = pathlib.Path(__file__).parent.resolve()
+    env = os.environ
+    _enable = os.environ.pop('NPY_ENABLE_CPU_FEATURES', None)
+    _disable = os.environ.pop('NPY_DISABLE_CPU_FEATURES', None)
+    SUBPROCESS_ARGS = dict(cwd=cwd, capture_output=True, text=True, check=True)
+    unavailable_feats = [
+        feat for feat in __cpu_dispatch__ if not __cpu_features__[feat]
+    ]
+    UNAVAILABLE_FEAT = (
+        None if len(unavailable_feats) == 0
+        else unavailable_feats[0]
+    )
+    BASELINE_FEAT = None if len(__cpu_baseline__) == 0 else __cpu_baseline__[0]
+
+    def _run(self):
+        path_to_np = os.path.dirname(np.__file__)
+        return subprocess.run(
+            [sys.executable, f'{path_to_np}/core/tests/_runtime_detect.py'],
+            **self.SUBPROCESS_ARGS,
+            )
+
+    # Helper function mimicing pytest.raises for subprocess call
+    def _expect_error(
+        self,
+        msg,
+        err_type,
+        no_error_msg="Failed to generate error"
+    ):
+        try:
+            self._run()
+        except subprocess.CalledProcessError as e:
+            assertion_message = f"Expected: {msg}\nGot: {e.stderr}"
+            assert re.search(msg, e.stderr), assertion_message
+
+            assertion_message = (
+                f"Expected error of type: {err_type}; see full "
+                f"error:\n{e.stderr}"
+            )
+            assert re.search(err_type, e.stderr), assertion_message
+        else:
+            assert False, no_error_msg
+
+    def setup_method(self, method):
+        """Ensure that neither variable is set unless specified in a test"""
+        self.env.pop('NPY_ENABLE_CPU_FEATURES', None)
+        self.env.pop('NPY_DISABLE_CPU_FEATURES', None)
```

Comment:
```suggestion
```
Any reasons forbid from passing the environment variable at new child process?
Both functions `subprocess.run` and  `subprocess.open` supports specifying custom environment variables.